### PR TITLE
Translate Ruby 2.4.0-preview1 (pt)

### PR DESCRIPTION
**Pull request by jcserracampos**

### CODE REVIEW

**Comment by JuanitoFatas** [comment]
cc @alanwillms & @fpgentil

File: pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+translator: "jcserracampos"
+date: 2016-06-20 18:00:00 +0900
+lang: pt
+---
+
+Estamos satisfeitos em anunciar o lançamento do Ruby 2.4.0-preview1.
+
+Ruby 2.4.0-preview1 é o primeiro *preview* do Ruby 2.4.0.
+Este preview1 está sendo lançado antes do usual porque ele inclui tantas 
+funcionalidades novas e aprimoramentos.
```
Comment (fpgentil) [inline pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: Based on other Portuguese posts, `improvements` has always been translated to `melhorias`, I would keep the same.

File: pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+Ruby 2.4 as unifica em Integer.
+
+## [String suporta mapeamento de caixa Unicode](https://bugs.ruby-lang.org/issues/10085)
+
+`String/Symbol#upcase/downcase/swapcase/capitalize(!)` agora lida
+com mapeamento de caixa Unicode ao invés de apenas mapeamento de caixa ASCII.
+
+## Aperfeiçoamento de perfomance
+
+Ruby 2.4 também contém os seguintes aprimoramentos de performance incluindo
```
Comment (fpgentil) [inline pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: ..also here in `aprimoramentos`

File: pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+`[x, y].max` e `[x, y].min` estão otimizadas para não criarem um *array* temporário
+dentro de certas condições.
+
+### [Regexp#match?](https://bugs.ruby-lang.org/issues/8110)
+
+Adicionada `Regexp#match?`, que executa uma combinação de expressão regular sem criar
+um objeto de referência de volta e mudando '$~' para reduzir a alocação de objeto.
+
+### Outros aprimoramentos de perfomance
```
Comment (fpgentil) [inline pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: and here (`aprimoramentos`).

File: pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+## Debugando
+
+### [Thread#report_on_exception e Thread.report_on_exception](https://bugs.ruby-lang.org/issues/6647)
+
+Ruby ignora exceções em *threads* a não ser que outra *thread* se junte explicitamente a ela.
+Com `report_on_exception = true`,
+você pode notar se uma *thread* morreu devido a uma exceção não processada.
+
+Envie-nos comentários sobre qual deve ser o padrão para `report_on_exception`
+e sobre report-on-GC.
```
Comment (fpgentil) [inline pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: I think *report-on-GC* should be in italic, does it make sense?

File: pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+* <https://cache.ruby-lang.org/pub/ruby/2.4/ruby-2.4.0-preview1.zip>
+
+  * SIZE:   19904781 bytes
+  * SHA1:   25c16ee91bbcb0224f80e20d938f5c08832973f7
+  * SHA256: fd588aea1558b1171f87a3dd342ee207b8995a2c0a8241d7aa15bcfa16036854
+  * SHA512: f2fff35ff9157a4b31177b3d6b91bdaad04c22b3c626c3a5e5ec9a31b103f9607b31c909ef27880065cfdbcfa5d6901a6db89d22e0c645666d069c5b6dd1818b
+
+## Comentário de lançamento
+
+Também veja calendário de lançamento e outras informações:
```
Comment (fpgentil) [inline pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: I think `Veja também o calendário de lançmento e outras informações:` sounds better.

File: pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md (hunk 4 above)
Comment (jcserracampos) [inline pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: It makes.

File: pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md (hunk 5 above)
Comment (jcserracampos) [inline pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: I agree.

File: pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+---
+
+Estamos satisfeitos em anunciar o lançamento do Ruby 2.4.0-preview1.
+
+Ruby 2.4.0-preview1 é o primeiro *preview* do Ruby 2.4.0.
+Este preview1 está sendo lançado antes do usual porque ele inclui tantas 
+funcionalidades novas e aprimoramentos.
+Sinta-se a vontade para 
+[enviar comentários](https://bugs.ruby-lang.org/projects/ruby/wiki/HowToReport) 
+desde que você ainda pode mudar as funcionalidades.
```
Comment (alanwillms) [inline pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: I agree @fpgentil

**Comment by stomar** [comment]
@jcserracampos Could you apply those suggestions?

@ whoever merges: please squash before merge.

**Comment by jcserracampos** [comment]
@stomar I'll fix today.
Sorry I was busy.

**Comment by jcserracampos** [comment]
@stomar Can you verify my new commit?
Sorry for delayed response.

**Comment by stomar** [comment]
@fpgentil

File: pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+author: "naruse"
+translator: "jcserracampos"
+date: 2016-06-20 18:00:00 +0900
+lang: pt
+---
+
+Estamos satisfeitos em anunciar o lançamento do Ruby 2.4.0-preview1.
+
+Ruby 2.4.0-preview1 é o primeiro *preview* do Ruby 2.4.0.
+Este preview1 está sendo lançado antes do usual porque ele inclui tantas
```
Comment (lucianosousa) [inline pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: `inclui muitas` ?

File: pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+funcionalidades novas e melhorias.
+Sinta-se a vontade para
+[enviar comentários](https://bugs.ruby-lang.org/projects/ruby/wiki/HowToReport)
+já que você ainda pode mudar as funcionalidades.
+
+## [Unificação de Fixnum e Bignum em Integer](https://bugs.ruby-lang.org/issues/12005)
+
+Embora [ISO/IEC 30170:2012](http://www.iso.org/iso/iso_catalogue/catalogue_tc/catalogue_detail.htm?csnumber=59579)
```
Comment (lucianosousa) [inline pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: `Embora a...`

File: pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+não especifique detalhes da classe Integer,
+CRuby tem duas classes visíveis de Integer: Fixnum e Bignum.
+Ruby 2.4 as unifica em Integer.
+
+## [String suporta mapeamento de caixa Unicode](https://bugs.ruby-lang.org/issues/10085)
```
Comment (lucianosousa) [inline pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: `%s/caixa/caso/g`

File: pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+
+`String/Symbol#upcase/downcase/swapcase/capitalize(!)` agora lida
+com mapeamento de caixa Unicode ao invés de apenas mapeamento de caixa ASCII.
```
Comment (lucianosousa) [inline pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: same from above

File: pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+
+## Aperfeiçoamento de perfomance
```
Comment (lucianosousa) [inline pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: most of the time, devs translate `Improvements` as `Melhorias`, not `Aperfeiçoamento`, but it's also a valid translation

File: pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+
+`[x, y].max` e `[x, y].min` estão otimizadas para não criarem um *array* temporário
+dentro de certas condições.
+
+### [Regexp#match?](https://bugs.ruby-lang.org/issues/8110)
+
+Adicionada `Regexp#match?`, que executa uma combinação de expressão regular sem criar
+um objeto de referência de volta e mudando '$~' para reduzir a alocação de objeto.
+
+### Outras melhorias de perfomance
```
Comment (lucianosousa) [inline pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: `melhorias` ;)

File: pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+
+* [acelerada o acesso a variável de instância](https://bugs.ruby-lang.org/issues/12274)
```
Comment (lucianosousa) [inline pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: `acelerado`, or just `acelerar` works more closest to the english language

File: pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+### [Thread#report_on_exception e Thread.report_on_exception](https://bugs.ruby-lang.org/issues/6647)
+
+Ruby ignora exceções em *threads* a não ser que outra *thread* se junte explicitamente a ela.
+Com `report_on_exception = true`,
+você pode notar se uma *thread* morreu devido a uma exceção não processada.
+
+Envie-nos comentários sobre qual deve ser o padrão para `report_on_exception`
+e sobre *report-on-GC*.
+
+### [Detecção de *thread deadlock* agora mostra quais *threads* com seu histórico e dependências](https://bugs.ruby-lang.org/issues/8214)
```
Comment (lucianosousa) [inline pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: `quais` can be removed. not obligatory but it's also another common sense in translations

**Comment by lucianosousa** [comment]
I did few comments here and I hope they might be useful
@jcserracampos any question feel free to ping me

File: pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md (hunk 7 above)
Comment (fpgentil) [inline pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: I even prefer `várias` than `muitas`, but I think both are better than `tantas`

File: pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md (hunk 9 above)
Comment (fpgentil) [inline pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: I tried to find any other references for `Unicode case mapping` but I couldn't. I checked Wikipedia and they say `mapeamento de código Unicode`, what do you guys think?

File: pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md (hunk 11 above)
Comment (fpgentil) [inline pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: I think `Melhorias` sounds better, doesn't it?

File: pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md (hunk 13 above)
Comment (fpgentil) [inline pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: If its `o acesso` the correct form is `acelerado`, but thinking about the original version I believe `acelera` goes even better than `acelerar`

File: pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md [hunk 15]
```diff
@@ -0,0 +1,115 @@
...
+
+* [acelerada o acesso a variável de instância](https://bugs.ruby-lang.org/issues/12274)
+
+## Debugando
+
+### [Thread#report_on_exception e Thread.report_on_exception](https://bugs.ruby-lang.org/issues/6647)
+
+Ruby ignora exceções em *threads* a não ser que outra *thread* se junte explicitamente a ela.
+Com `report_on_exception = true`,
+você pode notar se uma *thread* morreu devido a uma exceção não processada.
```
Comment (fpgentil) [inline pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: I think in this case `notar` is correct, these are false cognates

- `Notice [en]` means `notar [pt]`
- `Note [en]` means `notificar [pt]`

File: pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md (hunk 14 above)
Comment (fpgentil) [inline pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: I agree with the removal, it sounds kind of strange with `quais threads com seu histórico e dependência`, it seems to be missing something here, like `quais threads o quê?`

File: pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md [hunk 16]
```diff
@@ -0,0 +1,115 @@
...
+
+Envie-nos comentários sobre qual deve ser o padrão para `report_on_exception`
+e sobre *report-on-GC*.
+
+### [Detecção de *thread deadlock* agora mostra quais *threads* com seu histórico e dependências](https://bugs.ruby-lang.org/issues/8214)
+
+Ruby tem deteção de *deadlock* para *threads* em espera, mas seu relatório não
+inclue informações suficientes para debugar.
```
Comment (fpgentil) [inline pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: There's no `debugar` in the Portuguese language, I guess you could leave `*debugging*`

File: pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md [hunk 17]
```diff
@@ -0,0 +1,115 @@
...
+
+## Mudanças notáveis desde de 2.3
+
+Veja [NOTÍCIAS](https://github.com/ruby/ruby/blob/v2_4_0_preview1/NEWS)
+e [ChangeLog](https://github.com/ruby/ruby/blob/v2_4_0_preview1/ChangeLog)
+para detalhes.
+
+Com essas mudanças,
+[1140 arquivos mudaram, 33126 inserções(+), 50993 remoções(-)](https://github.com/ruby/ruby/compare/v2_3_0...v2_4_0_preview1)
+desde Ruby 2.3.0!
```
Comment (fpgentil) [inline pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: `Ruby` is always mentioned with the article, so `desde o Ruby 2.3.0`

File: pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md (hunk 9 above)
Comment (lucianosousa) [inline pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: @fpgentil makes sense for me

File: pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md (hunk 11 above)
Comment (lucianosousa) [inline pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: imho, yes. ;)

File: pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md (hunk 13 above)
Comment (jcserracampos) [inline pt/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md]: It's a new feature so it was implemented in this build.
I believe past tense is better.

**Comment by jcserracampos** [comment]
I'm having problemas squashing commits, I'll try again.

**Comment by jcserracampos** [comment]
One commit and corrections done.

**Pull request merged**